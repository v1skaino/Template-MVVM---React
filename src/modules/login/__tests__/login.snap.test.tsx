import renderer from 'react-test-renderer';
import { expect, it, vi } from 'vitest';
import { LoginModel } from '../model';
import LoginView from '../view';

vi.mock('../view.model', () => {
  const fnLoginViewModel = vi.fn();
  fnLoginViewModel.mockImplementation(
    () =>
      ({
        state: {
          email: '',
          password: '',
          loading: false,
        },
        handlers: {
          setEmail: vi.fn(),
          setPassword: vi.fn(),
          submit: vi.fn(),
        },
      } as LoginModel),
  );

  return { default: fnLoginViewModel };
});

import useLoginViewModel from '../view.model';

it('should render correctly', () => {
  const { toJSON } = renderer.create(<LoginView />);
  expect(toJSON()).toMatchSnapshot();
});

it('should render correctly with email', () => {
  useLoginViewModel.mockImplementation(() => ({
    state: {
      email: 'teste@email.com.br',
      password: '',
      loading: false,
    },
    handlers: {
      setEmail: vi.fn(),
      setPassword: vi.fn(),
      submit: vi.fn(),
    },
  }));

  const { toJSON } = renderer.create(<LoginView />);
  expect(toJSON()).toMatchSnapshot();
});

it('should render correctly with password', () => {
  useLoginViewModel.mockImplementation(() => ({
    state: {
      email: '',
      password: '123',
      loading: false,
    },
    handlers: {
      setEmail: vi.fn(),
      setPassword: vi.fn(),
      submit: vi.fn(),
    },
  }));

  const { toJSON } = renderer.create(<LoginView />);
  expect(toJSON()).toMatchSnapshot();
});

it('should render correctly with password and email', () => {
  useLoginViewModel.mockImplementation(() => ({
    state: {
      email: 'teste@email.com.br',
      password: '123',
      loading: false,
    },
    handlers: {
      setEmail: vi.fn(),
      setPassword: vi.fn(),
      submit: vi.fn(),
    },
  }));

  const { toJSON } = renderer.create(<LoginView />);
  expect(toJSON()).toMatchSnapshot();
});

it('should render while loading', () => {
  useLoginViewModel.mockImplementation(() => ({
    state: {
      email: 'teste@email.com.br',
      password: '123',
      loading: true,
    },
    handlers: {
      setEmail: vi.fn(),
      setPassword: vi.fn(),
      submit: vi.fn(),
    },
  }));

  const { toJSON } = renderer.create(<LoginView />);
  expect(toJSON()).toMatchSnapshot();
});
