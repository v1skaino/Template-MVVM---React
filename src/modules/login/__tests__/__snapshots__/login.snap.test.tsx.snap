// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should render correctly 1`] = `
<form
  onSubmit={[MockFunction spy]}
>
  <label>
    Email
  </label>
  <input
    onChange={[Function]}
    placeholder="user@email.com"
    value=""
  />
  <label>
    Senha
  </label>
  <input
    autoComplete="on"
    onChange={[Function]}
    placeholder="*****"
    type="password"
    value=""
  />
  <button
    disabled={false}
    type="submit"
  >
    ENTRAR
  </button>
</form>
`;

exports[`should render correctly with email 1`] = `
<form
  onSubmit={[MockFunction spy]}
>
  <label>
    Email
  </label>
  <input
    onChange={[Function]}
    placeholder="user@email.com"
    value="teste@email.com.br"
  />
  <label>
    Senha
  </label>
  <input
    autoComplete="on"
    onChange={[Function]}
    placeholder="*****"
    type="password"
    value=""
  />
  <button
    disabled={false}
    type="submit"
  >
    ENTRAR
  </button>
</form>
`;

exports[`should render correctly with password 1`] = `
<form
  onSubmit={[MockFunction spy]}
>
  <label>
    Email
  </label>
  <input
    onChange={[Function]}
    placeholder="user@email.com"
    value=""
  />
  <label>
    Senha
  </label>
  <input
    autoComplete="on"
    onChange={[Function]}
    placeholder="*****"
    type="password"
    value="123"
  />
  <button
    disabled={false}
    type="submit"
  >
    ENTRAR
  </button>
</form>
`;

exports[`should render correctly with password and email 1`] = `
<form
  onSubmit={[MockFunction spy]}
>
  <label>
    Email
  </label>
  <input
    onChange={[Function]}
    placeholder="user@email.com"
    value="teste@email.com.br"
  />
  <label>
    Senha
  </label>
  <input
    autoComplete="on"
    onChange={[Function]}
    placeholder="*****"
    type="password"
    value="123"
  />
  <button
    disabled={false}
    type="submit"
  >
    ENTRAR
  </button>
</form>
`;

exports[`should render while loading 1`] = `
<form
  onSubmit={[MockFunction spy]}
>
  <p>
    CARREGANDO...
  </p>
  <label>
    Email
  </label>
  <input
    onChange={[Function]}
    placeholder="user@email.com"
    value="teste@email.com.br"
  />
  <label>
    Senha
  </label>
  <input
    autoComplete="on"
    onChange={[Function]}
    placeholder="*****"
    type="password"
    value="123"
  />
  <button
    disabled={true}
    type="submit"
  >
    ENTRAR
  </button>
</form>
`;
